capstone/src/main/java/com/google/sps/servlets/student/StudentServlet.java

```diff
@@ -105,16 +112,54 @@ private Entity createStudentEntity(String userEmail) {
     return studentEntity;
   }
 
-  private ImmutableList<String> getAllAnnouncements(ImmutableList<String> clubNames) {
-    // TODO: Get announcements from Datastore once announcements have been loaded into Datastore
+  private ImmutableList<String> getAllAnnouncements(
+      ImmutableList<String> clubNames, DatastoreService datastore) {
     ImmutableList<String> announcements =
         Streams.stream(clubNames)
-            .flatMap(
-                clubName ->
-                    PrototypeClubs.PROTOTYPE_CLUBS_MAP.get(clubName).getAnnouncements().stream())
+            .flatMap(clubName -> getClubAnnouncements(clubName, datastore).stream())
             .collect(toImmutableList());
     return announcements;
   }
+
+  private ImmutableList<String> getClubAnnouncements(String clubName, DatastoreService datastore) {
+    // Get all announcements from given club name in reverse chronological order
+    Query query =
+        new Query(Constants.ANNOUNCEMENT_PROP)
+            .setFilter(new FilterPredicate(Constants.CLUB_PROP, FilterOperator.EQUAL, clubName))
+            .addSort(Constants.TIME_PROP, SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    // Stream through results and get formatted announcements
+    ImmutableList<String> announcements =
+        Streams.stream(results.asIterable())
+            .limit(Constants.LOAD_LIMIT)
+            .map(entity -> getAnnouncementAsString(entity))
+            .collect(toImmutableList());
+    return announcements;
+  }
+
+  private String getAnnouncementAsString(Entity announcement) {
+    // Set calendar timezone and time
+    TimeZone timePST = TimeZone.getTimeZone("PST");
```

Comment:
nit: constant (defined in this class is fine)

Reply:
Done!